### PR TITLE
Add basic HP3 info

### PR DESCRIPTION
**Pull request by syyyr**

### CODE REVIEW

File: src/history-provider.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+To do that _HP3_ does these things:
+- retrieves logs from other sources
+- exposes a `getLog` interface for every site
+- exposes a `_shvjournal` node, which serves as a controlling node and also as a container for every file that _HP3_ has
+  under its possession
+
+## Startup
+At startup, _HP3_:
+- Retrieves sites definition via `getSites`.
```
Comment (fvacek) [inline src/history-provider.md:12]: `sites/getSites`

File: src/history-provider.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+This is the most straightforward way of getting logs and isn't really "downloading". _HP3_ subscribes all the sites for
+`chng` events and saves them to a file called dirtylog. The dirtylog serves as a temporary storage of logs, until actual
+logs are retrieved from other sources. When new logs are retrieved, entries that are older than the newest synced entry
+are removed. This process is called "dirty log trimming".
+
+### File synchronization
+The standard way for retrieving logs is via file synchronization. _HP3_ retrieves a file list from a node and checks the
+difference between the local database and the remote database. All newer files are synced to the local database. The
+filelist can be provided by either the "device" (_shvgate_) or "HP3" and the node that contains the filelist is called
+`shvjournal` or `_shvjournal`. The location of the filelist node (and the file nodes) can be found through the sites
```
Comment (fvacek) [inline src/history-provider.md:69]: `.app/shvjournal`

File: src/history-provider.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+### Periodically over time
+_HP3_ iterates over all sites and syncs them periodically one by one. The interval between each site is configurable
+(default one minute).
+
+### On the `mntchng` signal
+Whenever a site gets a `mntchng` signal, it might mean that it has been offline for some time, and our data is stale,
+because we couldn't get events for our dirty log. Therefore, the site gets synced.
+
+## Sanitizer
```
Comment (fvacek) [inline src/history-provider.md:89]: default?

File: src/history-provider.md (hunk 1 above)
Comment (syyyr) [reply src/history-provider.md]: done

File: src/history-provider.md (hunk 2 above)
Comment (syyyr) [reply src/history-provider.md:73]: done

File: src/history-provider.md (hunk 3 above)
Comment (syyyr) [reply src/history-provider.md:94]: done

**Comment by syyyr** [comment]
This is ready.

**Pull request merged**